src/main/java/chatterbot/data/Deadline.java

```diff
@@ -0,0 +1,29 @@
+package chatterbot.data;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+
+    protected String by;
+
+    public Deadline(String description, String by) {
+        super(description);
+        this.by = by;
+    }
+
+    @Override
+    public String toString() {
+        return "[D]" + super.toString() + " (by: " + modifyBy(by) + ")";
+    }
+
+    public String modifyBy(String by) {
+        LocalDate ld = LocalDate.parse(by);
+        return ld.format(DateTimeFormatter.ofPattern("MMM d yyyy"));
+    }
+
+    @Override
```

Comment:
Could the method name be clearer if it was named as a verb? Maybe something like `convertToFileText`